### PR TITLE
Create/Delete Metro Volume Group

### PR DESCRIPTION
**Pull request by lukeatdell**
# Description
Adding support for `volume_group` endpoint actions, `configure_metro` and `end_metro` to configure and end metro replication for volume groups.
* Updated go modules.
* Updated mocks.
* Added unit and integration tests for `configure_metro` and `end_metro`.
* Improved metrics integration tests to avoid nil pointer dereference.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| --------------- |
| https://github.com/dell/csm/issues/1443 |

# Common PR Checklist:

- [x] Have you made sure that the code compiles?
- [x] Have you commented your code, particularly in hard-to-understand areas
- [x] Have you maintained backward compatibility

# Testing
## Integration Testing
+1.6%
image

## Unit Testing
+0.1%
image
+0.0%
image

### CODE REVIEW

File: volume_group.go [hunk 1]
```diff
@@ -297,3 +299,37 @@ func (c *ClientIMPL) GetVolumeGroupSnapshotByName(ctx context.Context, name stri
...
+			Body:     config,
+		},
+		&session)
+
+	return session, WrapErr(err)
+}
+
+// EndMetroVolumeGroup ends a metro configuration from a volume group and keeps both volume groups by default. The local copy
+// will retain its SCSI Identities while the remote volume group members will get new SCSI Identities if kept.
+func (c *ClientIMPL) EndMetroVolumeGroup(ctx context.Context, id string, config *EndMetroVolumeGroupOptions) (resp EmptyResponse, err error) {
```
Comment (santhoshatdell) [inline volume_group.go:322]: config -> options as param name?

File: inttests/volume_group_test.go [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+	// Create a metro vg session.
+	_, err = s.client.ConfigureMetroVolumeGroup(context.Background(), s.vg.this.ID, &s.metro.config)
+	if err != nil {
+		s.T().Skipf("Could not create metro volume group session. Skipping test... Err: %s", err)
+	}
+}
+
+func (s *EndMetroVolumeGroupTestSuite) TearDownTest() {
+	// End the metro session if one still exists.
+	s.client.EndMetroVolumeGroup(context.Background(), s.vg.this.ID, &s.metro.endOpts)
```
Comment (santhoshatdell) [inline inttests/volume_group_test.go:327]: Won't we get the below error for successful end-metro test (_TestEndMetroVolumeGroup_())?
"the volume group is not in a metro replication session"

**Review by santhoshatdell — DISMISSED** [review]

File: volume_group.go (hunk 1 above)
Comment (lukeatdell) [reply volume_group.go:322]: yep that makes sense, considering they are optional.

File: volume_types.go [hunk 3]
```diff
@@ -415,3 +415,15 @@ type EndMetroVolumeOptions struct {
 	// It is not recommended to use this option unless the remote side is known to be down.
 	ForceDelete bool `json:"force,omitempty"`
 }
+
+// EndMetroVolumeGroupOptions provides options for deleting the remote volume group and forcing the deletion.
+type EndMetroVolumeGroupOptions struct {
```
[the remark this answers is not in the document]
Comment (lukeatdell) [reply volume_types.go:420]: Oh yep. That makes sense. I'll move it.

File: inttests/volume_group_test.go (hunk 2 above)
Comment (lukeatdell) [reply inttests/volume_group_test.go:327]: Yes. That's why I've chosen not to bother checking the error response. We don't care if it fails here, we just want to clean up the PowerStore array and as a precursor to deleting the volumes and volume groups, we have to make sure it isn't still part of a replication session.

**Review by donatwork — APPROVED** [review]

**Review by santhoshatdell — APPROVED** [review]

**Pull request merged**